LoopBack 2.x branch test failures

<!--
Questions:
  https://groups.google.com/forum/#!forum/loopbackjs
  https://gitter.im/strongloop/loopback
Immediate support:
  https://strongloop.com/api-connect-faqs/
  https://strongloop.com/node-js/subscription-plans/
-->

# Description/Steps to reproduce

2 tests are failing on `2.x` branch. Run `npm it`.

## Failing Tests

```
  1) loopback loopback.getCurrentContext preserves callback domain:
      Uncaught AssertionError: expected 'context not available' to equal 'a value stored in context'
      + expected - actual
      -context not available
      +a value stored in context
      
      at Test.<anonymous> (test/loopback.test.js:629:36)
      at Test.assert (node_modules/supertest/lib/test.js:179:6)
      at Server.assert (node_modules/supertest/lib/test.js:131:12)
      at emitCloseNT (net.js:1659:8)
      at _combinedTickCallback (internal/process/next_tick.js:135:11)
      at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:218:9)
  2) loopback.rest context propagation should enable context with loopback.rest:
     Error: expected 200 "OK", got 500 "Internal Server Error"
      at Test._assertStatus (node_modules/supertest/lib/test.js:266:12)
      at Test._assertFunction (node_modules/supertest/lib/test.js:281:11)
      at Test.assert (node_modules/supertest/lib/test.js:171:18)
      at Server.assert (node_modules/supertest/lib/test.js:131:12)
      at emitCloseNT (net.js:1659:8)
      at _combinedTickCallback (internal/process/next_tick.js:135:11)
      at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:218:9)
```

# Expected result

<!--
Also include actual results if bug
-->

All tests should pass.

--- a/server/middleware/rest.js
+++ b/server/middleware/rest.js
@@ -73,8 +73,21 @@ function rest() {
     if (handlers.length === 1) {
       return handlers[0](req, res, next);
     }
-    async.eachSeries(handlers, function(handler, done) {
-      handler(req, res, done);
-    }, next);
+
+    executeHandlers(handlers, req, res, next);
   };
+}
+
+// A trimmed-down version of async.series that preserves current CLS context
+function executeHandlers(handlers, req, res, cb) {
+  var ix = -1;
+  next();
+
+  function next(err) {
+    if (err || ++ix >= handlers.length) {
+      cb(err);
+    } else {
+      handlers[ix](req, res, next);
+    }
+  }
 }

